src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -5712,6 +5768,9 @@ QGISEXTERN QString loadStyle( const QString &uri, QString &errCause )
   QgsDataSourceUri dsUri( uri );
   QString sqlitePath = dsUri.database();
   QgsDebugMsg( "Database is: " + sqlitePath );
+  // Avoid sql-statement errors
+  if ( dsUri.schema().isEmpty() )
+    return false;
```

Comment:
Sorry, made the correction in the wrong copy.
The 'else' (from above) has been added and QString()  returned for both cases with a message for the reason.